### PR TITLE
feat: SITES-9963 adding invalidate cache logic

### PR DESCRIPTION
**Pull request by tyge68**
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues
SITES-9963 In order to demo the full flow , it need to send a cache invalidation to cloudflare

Thanks for contributing!

### CODE REVIEW

**Comment by tyge68** [comment]
@tripodsan could you check this change, then let me know if I can merge it to use a stable url on Odin QA (and soon Odin Prod)

File: src/request-util.js [hunk 1]
```diff
@@ -29,13 +34,21 @@ export default class RequestUtil {
...
-      this.json = await this.request.json();
+      this.text = await this.request.text();
+      if (this.text.indexOf(BASE64_HEADER) === 0) {
+        // decode base64 encoded for large json
+        const base64part = this.text.substring(BASE64_HEADER.length);
+        const data = await gunzip(Buffer.from(base64part, 'base64'));
+        this.json = JSON.parse(data.toString('utf8'));
+      } else {
+        this.json = JSON.parse(this.text);
+      }
```
Comment (tripodsan) [inline src/request-util.js:51]: but this is not correct. when the content-type is json, it must not be base64 encoded and zipped.
also, why not sending binary data? base64 makes the payload 20% larger.
If you want to support zipped payloads, I would keep the content-type app/json, but support transport-encoding: gzip. this way it's all nicely abstracted via the http protocol.

File: src/request-util.js [hunk 2]
```diff
@@ -29,13 +34,21 @@ export default class RequestUtil {
       return;
     }
 
-    if (this.request.headers.get('Content-Type') !== 'application/json') {
+    if (this.request.headers.get('Content-Type') !== APPLICATION_JSON) {
```
Comment (tripodsan) [inline src/request-util.js:37]: this check is very strict, eg in respect to whitespaces. I would do `startsWith(...)`.

**Review by tripodsan — CHANGES_REQUESTED** [review]
don't confuse the API user with weird payload handling

File: src/request-util.js (hunk 1 above)
Comment (tyge68) [reply src/request-util.js:51]: @tripodsan make sense I will check if that can be done / tested also.

File: src/request-util.js [hunk 3]
```diff
@@ -29,13 +34,19 @@ export default class RequestUtil {
       return;
     }
 
-    if (this.request.headers.get('Content-Type') !== 'application/json') {
+    if (!this.request.headers.get('Content-Type').startsWith(APPLICATION_JSON)) {
```
Comment (tripodsan) [inline src/request-util.js:37]: but now this is wrong. if should only startWith(`application/json`) not with the encoding

**Pull request merged**